loopy/schedule/__init__.py

```diff
@@ -1649,7 +1649,7 @@ def filter_var_set_for_base_storage(var_name_set, base_storage_name):
                     race_var, = src_race_vars
 
                     if not (
-                        self.overlap_checker.do_access_ranges_overlap_conservative(
+                        self.write_race_checker.do_accesses_result_in_races(
```

Comment:
Should this just outright replace `AccessRnageOverlapChecker`?

Reply:
`check_variable_access_ordered` also uses AccessRangeOverlapChecker. I think that's a genuine use-case for the mapper.

Reply:
But... `check_variable_access_ordered` also checks for potential races. Are these two bits of functionality redundant?

Reply:
> check_variable_access_ordered also checks for potential races

To the best of my knowledge, it doesn't. It just grabs the range of the access map over an insn's within inames.

I think you might be hinting at `check_for_write_races` which is quite imprecise in its check. Warns about the write-races by checking dependencies on hw inames (doesn't even look at access maps) and does so by looking solely at an instruction's local accesses (i.e. doesn't look at access pairs).

Reply:
FWIW, rewiring the current functionality of `check_for_write_races` to use `WriteRaceChecker` shouldn't be too disruptive.

Reply:
Can't quite tell what you're trying to say: Do that now, here? Or file an issue (to avoid forgetting) and do it later?

Reply:
Yup, I had intended this to come later. See #449.